replicatedhq/ship
# Eks in a box

What I Did
------------

Added an asset `amazon_elastic_kubernetes_service` that generates a terraform file that creates an EKS cluster.
A ton of Makefile changes to make the build/test/docs process easier.
The autogenerated docs now include child parameters.

How I Did it
------------

1.  `render.go` within the `amazonElasticKubernetesService` package generates terraform based on the options provided.
1.  The Makefile within `web` no longer reruns webpack unless files have changed. The top-level Makefile uses it as a dependency for `pkg/lifeycle/daemon/ui.bindatafs.go`. 
1. The documentation autogeneration script now also iterates through child parameters.

How to verify it
------------

1. Look at the added integration test, which generates the terraform scripts.
2. `git clean` the repository and `make test` and observe that the `pkg/lifeycle/daemon/ui.bindatafs.go` step no longer fails.
3. Run `make pipeline-strict` inside `hack/docs` and look at the generated file `assets/amazon_elastic_kubernetes_service.md`. `autoscaling_groups`, `created_vpc` and `existing_vpc` all have child parameters listed.

Description for the Changelog
------------

Added an asset `amazon_elastic_kubernetes_service` that generates a terraform file that creates an EKS cluster.


Picture of a ~Boat~Ship (not required but encouraged)
------------

![image](https://user-images.githubusercontent.com/2318911/43429265-4cef6e7e-9417-11e8-8fdd-83055cafaf54.png)









TODO:
1. Add this object to schema within [replicated-lint](https://github.com/replicatedhq/replicated-lint/tree/master/projects/replicated-ship)
1. Update documentation within [help-center](https://github.com/replicatedhq/help-center/tree/master/content/api/ship-assets)
1. Add integration test to console
